main/streams/streams.c

```diff
@@ -2332,6 +2332,29 @@ PHPAPI int _php_stream_scandir(const char *dirname, zend_string **namelist[], in
 }
 /* }}} */
 
+/* {{{ php_stream_is_cacheable
+ */
+PHPAPI int _php_stream_is_cacheable(const char *path, int options, php_stream_context *context)
+{
+	php_stream_wrapper *wrapper = NULL;
+	int ret;
+
+	if (!path || !*path) {
+		return 0;
+	}
+
+	wrapper = php_stream_locate_url_wrapper(path, NULL, options);
+
+	if (wrapper && wrapper->wops->stream_is_cacheable) {
+		ret=wrapper->wops->stream_is_cacheable(wrapper,
+				path, options, context);
+
+	} else ret=0;
```

Comment:
As for the coding standards https://github.com/php/php-src/blob/master/CODING_STANDARDS#L208

I would prefer: 
```
if () {
  ret = xxx;
} else {
  ret = 0;
}
```